### PR TITLE
 Add location to the attendance screen

### PR DESCRIPTION
**Pull request by sameera222**
This Pull Request Will 
- [x] Add current location to the Dataman App

### CODE REVIEW

File: apps/Dataman/src/screens/Location.tsx [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+  const [errorMsg, setErrorMsg] = useState("");
+  const [locationServiceEnabled, setLocationServiceEnabled] = useState(false);
+  const [displayCurrentAddress, setDisplayCurrentAddress] = useState(
+    "Wait, we are fetching you location..."
+  );
+
+  useEffect(() => {
+    CheckIfLocationEnabled();
+    GetCurrentLocation();
+  }, []);
```
Comment (sutyum) [inline apps/Dataman/src/screens/Location.tsx]: empty dependency array is a bad practice

File: apps/Dataman/src/screens/Location.tsx [hunk 2]
```diff
@@ -0,0 +1,60 @@
+import React, { useState, useEffect } from "react";
+import { Text, View } from "react-native";
+import * as Location from "expo-location";
+
+export const LocationScreen = () => {
+  const [errorMsg, setErrorMsg] = useState("");
+  const [locationServiceEnabled, setLocationServiceEnabled] = useState(false);
```
Comment (sutyum) [inline apps/Dataman/src/screens/Location.tsx]: you should refactor this entire block of code into a custom hook

**Review by sutyum — CHANGES_REQUESTED** [review]

File: apps/Dataman/src/screens/Location.tsx (hunk 1 above)
Comment (sameera222) [reply apps/Dataman/src/screens/Location.tsx]: now I have given the array dependency when the button is clicked the state will change.

File: apps/Dataman/src/screens/Location.tsx (hunk 2 above)
Comment (sameera222) [reply apps/Dataman/src/screens/Location.tsx]: refactored this code.

File: apps/Dataman/src/components/useLocation.tsx [hunk 3]
```diff
@@ -0,0 +1,49 @@
+import React, { useState, useEffect } from "react";
```
Comment (sutyum) [inline apps/Dataman/src/components/useLocation.tsx]: this file does not need to be .tsx -> .ts is better

Comment (sutyum) [reply apps/Dataman/src/components/useLocation.tsx]: Move to src/hooks/...

File: apps/Dataman/src/components/useLocation.tsx [hunk 4]
```diff
@@ -0,0 +1,49 @@
+import React, { useState, useEffect } from "react";
+import * as Location from "expo-location";
+
+export const useLocation = () => {
+  const [errorMsg, setErrorMsg] = useState("");
+  const [locationServiceEnabled, setLocationServiceEnabled] = useState(false);
```
Comment (sutyum) [inline apps/Dataman/src/components/useLocation.tsx]: for a boolean value prefer names such as `isLocationServiceEnabled`

File: apps/Dataman/src/components/useLocation.tsx [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+    const { coords } = await Location.getCurrentPositionAsync();
+
+    if (coords) {
+      const { latitude, longitude } = coords;
+      const response = await Location.reverseGeocodeAsync({
+        latitude,
+        longitude,
+      });
+
+      for (const item of response) {
```
Comment (sutyum) [inline apps/Dataman/src/components/useLocation.tsx]: why a loop is needed? Why not simply return the first element in the reponse

**Review by sutyum — CHANGES_REQUESTED** [review]

File: apps/Dataman/src/components/useLocation.tsx (hunk 5 above)
Comment (sameera222) [reply apps/Dataman/src/components/useLocation.tsx]: response is returning array of object that's why I used the loop. But now changed it and set directly.

File: apps/Dataman/src/components/useLocation.tsx (hunk 4 above)
Comment (sameera222) [reply apps/Dataman/src/components/useLocation.tsx]: done

File: apps/Dataman/src/components/useLocation.tsx (hunk 3 above)
Comment (sameera222) [reply apps/Dataman/src/components/useLocation.tsx]: done

**Review by sutyum — APPROVED** [review]

**Pull request merged**